### ISSUE TITLE
Router parameters default value equals to position

### ISSUE THREAD

**Issue by Surt** [phalcon/cphalcon#10428]
Using named parameters on the standard router I get a default value equal to the position, when it's not defined:

```php
    $router->addGet("/api/:module/([a-zA-Z0-9_-]+)(/json|/csv)*", array(
        'module' => 1,
        'controller' => 'index',
        'action' => 'view',
        'element' => 2,
        'type' => 3,
        'api' => true
    ));
```

using, on the controller: 

```php
$dispatcher->getParam('type')
```
```
api/user/1 ==> 3 // I don't want the 3 to be shown, the parameter is not set

api/user/1/json ==> '/json'
```

The problem comes when I try to parse the "type" parameter, I must check it vs the value 3, but in other routes I will have it as the second parameter so....

How can I accomplish this? Maybe there is another way to do it?

Also, notice that the "api" value is setted in a strange way: 

```
api => user
``` 

I did a discussion post some time ago:
http://forum.phalconphp.com/discussion/6674/router-default-for-named-parameters-at-the-end-of-the-route#C18867

**Comment by Surt** [comment]
Test for this:

```php
$testRoutes = array(
            '/api/user/rer',
            '/api/user',
            '/api/user/rer/json'
        );

        $router = new \Phalcon\Mvc\Router(false);

        $router->add("/api/:module/([a-zA-Z0-9_-]+)(/json|/csv)*", array(
            'module' => 1,
            'controller' => 'index',
            'action' => 'view',
            'element' => 2,
            'type' => 3,
            'api' => true
        ));
        foreach ($testRoutes as $testRoute) {

            //Handle the route
            $router->handle($testRoute);

            echo 'Testing ', $testRoute, '<br>';

            //Check if some route was matched
            if ($router->wasMatched()) {
                echo 'Module: ', $router->getModuleName(), '<br>';
                echo 'Controller: ', $router->getControllerName(), '<br>';
                echo 'Action: ', $router->getActionName(), '<br>';
                echo '<pre>Params: ', print_r($router->getParams(), true), '<br>';
            } else {
                echo 'The route wasn\'t matched by any route<br>';
            }
            echo '<br>';
        }
```

**Comment by andresgutierrez** [comment]
Closed as duplicate of #1078
